src/App.tsx

```diff
@@ -1,35 +1,29 @@
 import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import Counter from './components/counter';
+import Header from './components/header/header';
+import './App.css';
 
 function App() {
-  const [count, setCount] = useState(0)
+  let genreList=["Horror","Romantic","Thrilling", "Comedy"];
+  const [selectedGenre, setSelectedGenre] = useState('Horror');
 
+  const onSelectGenre = (genre:string) => () => {
+    console.log(genre);
+    setSelectedGenre(genre);
+}
   return (
     <>
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://react.dev" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
+        <Counter initialValue={1} /> 
+        <div className= "pageLayout">
+         <Header genreList={genreList} selectedGenre={selectedGenre} onSelect = {onSelectGenre}></Header>
```

Comment:
What is the reason of props drillig in this case? 
App (genrelist, selectedGenre, onSectedGenre) => Herader (dont use any of that props) => GenreSelect (uses genrelist, selectedGenre, onSectedGenre).
With your components composition, 'genrelist, selectedGenre, onSectedGenre' should live in the Header.

Reply:
i did it this way because in the coming task with is displaying the movies list corresponding to the genre selected so we will need to level it up away from the header to the app

Reply:
No sure that I undestand you correctly.